snmp/datadog_checks/snmp/config.py

```diff
@@ -108,16 +109,18 @@ def __init__(self, instance, warning, log, global_metrics, mibs_path, profiles,
 
         self._context_data = hlapi.ContextData(*self.get_context_data(instance))
 
+        if ip_address:
+            self._addr_name, _ = lcd.configure(
+                self._snmp_engine, self._auth_data, self._transport, self._context_data.contextName
+            )
+
     def resolve_oid(self, oid):
         return self._resolver.resolve_oid(oid)
 
     def refresh_with_profile(self, profile, warning, log):
         self.metrics.extend(profile['definition']['metrics'])
         self.all_oids, self.bulk_oids, self.parsed_metrics = self.parse_metrics(self.metrics, warning, log)
 
-    def call_cmd(self, cmd, *args, **kwargs):
-        return cmd(self._snmp_engine, self._auth_data, self._transport, self._context_data, *args, **kwargs)
```

Comment:
I like that this refactor reduces the amount of smarts on the `Config` class. I think we should encourage more of that going forward.

Reply:
You're right, but the reason was that it uses private attributes, so it's connected to your next comments.